### DISCUSSION TITLE
PDF bookmark

### DISCUSSION THREAD

**Discussion by pdml-director** [fmarotta/kaobook#259] [General]
I found this template and I love the design. However, when opening up the PDF in a PDF reader (e.g., adobe reader), the bookmarks are not correctly aligned/structured. How can we fix this problem?

**Comment by Artaraz** [comment]
I had the same concern, but I have started to use it for the thesis anyway because it's a really beautiful template. 
I've spent the whole day "chatting (and procastinating) with ChatGPT to see if this issue could be addressed easily... after a whole day and huge number of tries I got a straightforward solution from chatGPT: 
You just have to replace the "\newcommand{\mtocsection}[1]{...}" and "\newcommand{\mtocsection}[1]{ ...}" from the kao.sty file with the following:

`
\newcommand{\mtocsection}[1]{
    \hypersetup{bookmarksdepth=-2} % Temporarily suppress bookmarks %added%
    \addcontentsline{toc}{mtocsection}{%
        \ifnum\value{secnumdepth}>0 \protect\numberline{\thesection}%
        \else \protect\nonumberline%
        \fi #1}%
    \hypersetup{bookmarksdepth=2} % Restore bookmark levels %added%
}

\newcommand{\mtocsubsection}[1]{
    \hypersetup{bookmarksdepth=-2} % Temporarily suppress bookmarks %added%
    \addcontentsline{toc}{mtocsubsection}{%
        \ifnum\value{secnumdepth}>1 \protect\numberline{\thesubsection}%
        \else \protect\nonumberline%
        \fi #1}%
    \hypersetup{bookmarksdepth=2} % Restore bookmark levels %added%
}
`

It's basically just adding two lines for each newcommand (noted as %added%)

**Reply by pdml-director** [reply]
Thank you so much. This fixed the problem!!

**Reply by dirodgu** [reply]
Thank you very much! I was also stuck with the same problem :)

**Reply by kleinerELM** [reply]
This should be a pull request to fix the issue in the template for good.
Thanks!

**Reply by marcussporring** [reply]
Amazing! This saved me heaps of time:D

**Comment by MemoJimenez** [comment]
I had the same problem. I tried the code shared by @Artaraz (thank you!), but unfortunately, it didn't work for me. After experimenting with several solutions suggested by Claude, I found a solution and even organized my bookmarks by chapter!!!

<img width="200" height="500" alt="image" src="https://github.com/user-attachments/assets/f1dc4510-b035-4c21-b6f0-0ba7f38e4d58" />

Here’s the code I added right after \documentclass in my main.tex file:

`\makeatletter
\renewcommand{\mtocsection}[1]{%
    \begingroup
    \hypersetup{bookmarksdepth=-2}%
    \addcontentsline{toc}{mtocsection}{%
        \ifnum\value{secnumdepth}>0 \protect\numberline{\thesection}%
        \else \protect\nonumberline%
        \fi #1}%
    \endgroup
}

\renewcommand{\mtocsubsection}[1]{%
    \begingroup
    \hypersetup{bookmarksdepth=-2}%
    \addcontentsline{toc}{mtocsubsection}{%
        \ifnum\value{secnumdepth}>1 \protect\numberline{\thesubsection}%
        \else \protect\nonumberline%
        \fi #1}%
    \endgroup
}
\makeatother`


There was one small issue that I couldn’t resolve through programming — the bookmark for my cover page was blank. I tried countless approaches to force a “Cover” entry in the PDF bookmarks, but nothing worked. Finally, I used PDF-XChange Editor to rename the bookmark directly in the PDF to get this final touch.

Hope this helps someone else!
